### PR TITLE
Docs: make Writing Tests a bit more focused

### PR DESCRIPTION
**Pull request by annevk**
The goal of this change is to make this page more useful at a glance and for recurring visitors as well as for those not familiar with the testharness.js terminology.

### CODE REVIEW

**Comment by annevk** [comment]
So based on the build results I have a question. Is there a way to separate the TOC from the inline links?

File: docs/writing-tests/index.md [hunk 1]
```diff
@@ -8,43 +8,13 @@ testing plan](making-a-testing-plan) will help you decide what to write.
...
-   github-intro
-```
-
-## Test Type
+There are various different ways of writing tests:
 
-Tests in this project use a few different approaches to verify expected
-behavior. The tests can be classified based on the way they express
-expectations:
+* [JavaScript tests (testharness.js)](testharness) are preferred and should be used where possible.
```
Comment (jgraham) [inline docs/writing-tests/index.md:15]: I think `are preferred for testing DOM APIs and may be used for testing other features` if we want to talk about preferences.

**Review by sideshowbarker — APPROVED** [review]
LGTM

File: docs/writing-tests/index.md [hunk 2]
```diff
@@ -8,43 +8,14 @@ testing plan](making-a-testing-plan) will help you decide what to write.
 
 There's also a load of [general guidelines](general-guidelines) that apply to all tests.
 
-```eval_rst
-.. toctree::
-   :maxdepth: 1
+## Choose Your Category
```
Comment (jgraham) [inline docs/writing-tests/index.md:11]:
```suggestion
## Test Types
```

**Review by jgraham — APPROVED** [review]

**Pull request merged**